pyright_reportGeneralTypeIssues.json

```diff
@@ -1,4 +1,3 @@
-# this becomes obsolete when reportGeneralTypeIssues can be enabled in pyproject.toml
```

Comment:
Had to remove these comments as the JSON format does not allow comments (pyright seems to use a stricter parser now)

Reply:
what happens if you dont upgrade pyright?

Reply:
Upgrading pyright is not needed for this PR. Just upgraded it because there is a newer version.

Reply:
I reverted this change for this PR as it is not yet needed.